src/webgpu/api/operation/memory_sync/buffer.spec.ts

```diff
@@ -0,0 +1,261 @@
+  ): GPUBindGroup {
+    return this.device.createBindGroup({
+      layout: pipeline.getBindGroupLayout(0),
+      entries: [{ binding: 0, resource: { buffer } }],
+    });
+  }
+
+  // Create a compute pipeline and write given data into storage buffer.
+  createStorageWriteComputePipeline(value: number): GPUComputePipeline {
+    const wgslCompute = `
+      type Data = [[block]] struct {
+        [[offset 0]] a : i32;
+      };
+
+      [[binding 0, set 0]] var<storage_buffer> data : Data;
+      fn main() -> void {
+	data.a = ${value};
+        return;
+      }
+
+      entry_point compute = main;
+    `;
+
+    return this.device.createComputePipeline({
+      computeStage: {
+        module: this.device.createShaderModule({
+          code: wgslCompute,
+        }),
+        entryPoint: 'main',
+      },
+    });
+  }
+
+  // Create a render pipeline and write given data into storage buffer at fragment stage.
+  createStorageWriteRenderPipeline(value: number): GPURenderPipeline {
+    const wgslVertex = `
+    [[builtin position]] var<out> Position : vec4<f32>;
+      fn vert_main() -> void {
+        Position = vec4<f32>(0.5, 0.5, 0.0, 1.0);
+        return;
+      }
+
+      entry_point vertex = vert_main;
+    `;
+
+    const wgslFragment = `
+    [[location 0]] var<out> outColor : vec4<f32>;
+      type Data = [[block]] struct {
+        [[offset 0]] a : i32;
+      };
+
+      [[binding 0, set 0]] var<storage_buffer> data : Data;
+      fn frag_main() -> void {
+	data.a = ${value};
+	outColor = vec4<f32>(1.0, 0.0, 0.0, 1.0);
+        return;
+      }
+
+      entry_point fragment = frag_main;
+    `;
+
+    return this.device.createRenderPipeline({
+      vertexStage: {
+        module: this.device.createShaderModule({
+          code: wgslVertex,
+        }),
+        entryPoint: 'vert_main',
+      },
+      fragmentStage: {
+        module: this.device.createShaderModule({
+          code: wgslFragment,
+        }),
+        entryPoint: 'frag_main',
+      },
+      primitiveTopology: 'point-list',
+      colorStates: [{ format: 'rgba8unorm' }],
+    });
+  }
+
+  beginSimpleRenderPass(encoder: GPUCommandEncoder): GPURenderPassEncoder {
+    const view = this.device
+      .createTexture({
+        size: { width: 4, height: 4, depth: 1 },
+        format: 'rgba8unorm',
+        usage: GPUTextureUsage.OUTPUT_ATTACHMENT,
+      })
+      .createView();
+    return encoder.beginRenderPass({
+      colorAttachments: [
+        {
+          attachment: view,
+          loadValue: { r: 0.0, g: 1.0, b: 0.0, a: 1.0 },
+        },
+      ],
+    });
+  }
+
+  // Write storage buffer via draw call in render pass. Use bundle if needed.
+  writeByRenderPass(bundle: boolean, buffer: GPUBuffer, value: number, encoder: GPUCommandEncoder) {
+    const pipeline = this.createStorageWriteRenderPipeline(value);
+    const bindGroup = this.createBindGroup(pipeline, buffer);
+    const pass = this.beginSimpleRenderPass(encoder);
+    if (bundle) {
```

Comment:
Is it possible to use the helper from https://github.com/gpuweb/cts/pull/293?

Reply:
Hmm, I need to test different `scopes`, so I don't need to use `finish` in that function, which always suppose that the test need a separate command encoder (it is not true for this test). So it will become unused variable if I use that function.